File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int size=nums.size();
        int MaxSum=nums[0];
        int sum=0;
        for(int i=0;i<size;i++){
            sum+=nums[i];
            MaxSum=max(MaxSum,sum);
            if(sum<0){
                sum=0;
            }
        }
        return MaxSum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        maxsum=-float('inf')
        sum=0
        for i in range(len(nums)):
            sum+=nums[i]
            maxsum=max(maxsum,sum)
            if sum<0:
                sum=0
        return maxsum

            